packages/m2m-gateway/src/utils/fileUpload.ts

```diff
@@ -0,0 +1,78 @@
+import { catalogApi, m2mGatewayApi } from "pagopa-interop-api-clients";
+import {
+  FileManager,
+  Logger,
+  verifyAndCreateDocument,
+} from "pagopa-interop-commons";
+import { generateId, technology } from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { config } from "../config/config.js";
+import { CatalogProcessClient } from "../clients/clientsProvider.js";
+import { WithMaybeMetadata } from "../clients/zodiosWithMetadataPatch.js";
+import { Headers } from "./context.js";
+
+export async function uploadEServiceDocument({
+  eservice,
+  descriptorId,
+  documentKind,
+  fileUpload,
+  catalogProcessClient,
+  fileManager,
+  logger,
+  headers,
+}: {
+  eservice: catalogApi.EService;
+  descriptorId: string;
+  documentKind: catalogApi.EServiceDocumentKind;
+  fileUpload: m2mGatewayApi.FileUploadMultipart;
+  catalogProcessClient: CatalogProcessClient;
+  fileManager: FileManager;
+  logger: Logger;
+  headers: Headers;
+}): Promise<WithMaybeMetadata<catalogApi.EServiceDoc>> {
+  return await verifyAndCreateDocument(
```

Comment:
I checked the code in `verifyAndCreateDocument` and found a behaviour that I don't fully understand...
Shouldn't this piece of code be done only in case of "INTERFACE" and not in case of "DOCUMENT"?
Only if it's the interface, we should check the interface content type and extract serverUrls 🤔 

<img width="922" height="408" alt="image" src="https://github.com/user-attachments/assets/5cbaf6e1-8ad5-49aa-b24c-b7c4c7fd6ab1" />

Reply:
Will be addressed in #2227